libindy/src/services/crypto/mod.rs

```diff
@@ -43,22 +43,60 @@ impl CryptoService {
         }
     }
 
-    pub fn create_key(&self, key_info: &KeyInfo) -> Result<Key, CryptoError> {
-        let crypto_type_name = key_info.crypto_type
-            .as_ref()
-            .map(String::as_str)
-            .unwrap_or(DEFAULT_CRYPTO_TYPE);
+    pub fn get_crypto_type_name<'a>(crypto_type: &'a Option<String>) -> (&'a str) {
```

Comment:
I like the idea of reducing boilerplate, but we need to follow some rules here:

1. Each function should have a single responsibility. 
2. Each function should have obvious responsibility and a good name
3. We shouldn't expose private functions
4. Private functions should have a name started with '_'

Reply:
> pub fn get_crypto_type_name<'a>(crypto_type: &'a Option<String>) -> (&'a str)

It seems like non-obvious function and i suggest to remove it